src/skin/legacyskinparser.cpp

```diff
@@ -488,6 +489,10 @@ QList<QWidget*> LegacySkinParser::parseNode(QDomElement node) {
         m_pContext->updateVariable(node);
     } else if (nodeName == "Template") {
         result = parseTemplate(node);
+    } else if (nodeName == "DefineSingleton") {
+        parseSingletonDefinition(node);
+    } else if (nodeName == "Singleton") {
+        result = wrapWidget(parseSingletonContainer(node));
```

Comment:
The names are OK, I have just suspected it the other way round. We have no verbs in the other tag names. So you may think about this: 

<Singleton>  Where the singleton is instantiated. This would match with the other widgets like <Pushbutton> which is instantiated when named.  Alternative <SingletonInstance>  
And <SingletonContainer> or <SingeltonView> for the different views. 

Have you also thought about to get rid of the predefinition and just add a singleton property to the templates? 
This might be not that flexible, but much easier to use. 
<Template singleton= true>